storage-proofs/src/piece_inclusion_proof.rs

```diff
@@ -15,15 +16,57 @@ pub struct PieceInclusionProof<H: Hasher> {
     proof_elements: Vec<H::Domain>,
 }
 
+impl<H: Hasher> From<PieceInclusionProof<H>> for Vec<u8> {
+    fn from(proof: PieceInclusionProof<H>) -> Self {
+        let position = proof.position.to_le_bytes();
+        let proof_elements = proof
+            .proof_elements
+            .iter()
+            .flat_map(H::Domain::into_bytes)
+            .collect::<Vec<u8>>();
+
+        [&position, proof_elements.as_slice()].concat()
+    }
+}
+
+impl<H: Hasher> From<Vec<u8>> for PieceInclusionProof<H> {
```

Comment:
TODO: Replace `From<Vec<u8>> for PieceInclusionProof<H>` with a function called `try_into_piece_inclusion_proof` of type `&[u8] -> Option<PieceInclusionProof>`.